app/models/concerns/has_a_onestop_id.rb

```diff
@@ -5,28 +5,47 @@ module HasAOnestopId
     validates :onestop_id, presence: true, uniqueness: true
     validate :onestop_id, :validate_onestop_id
 
+    def self.current_from_old_reference(old_entity, find_method)
+      unless ['merge', 'change_onestop_id'].exclude?(old_entity.action) || old_entity.current.nil?
+        self.send(find_method, { onestop_id: old_entity.current.onestop_id })
+      end
+    end
+
     def self.find_by_onestop_id!(onestop_id)
```

Comment:
Split into 6 methods total:
Keeping the 2 previously implemented find_by_onestop_id methods
2 new methods named "find_by_current_and_old_onestop_id" (add comments to explain)
2 "plural" find_by_onestop_ids methods that maintain the existing behavior